src/main/com/fulcrologic/fulcro/raw/components.cljc

```diff
@@ -797,6 +797,21 @@
            ast-nodes (-> query eql/query->ast :children)]
        (get-subquery-component* component ast-nodes query-path)))))
 
+(defn get-traced-props
+  "Uses `fdn/traced-db->tree` to get the props of the component at `ident`. If `prior-props` are not stale,
+   those are returned instead."
+  [state-map component {:keys [ident prior-props]}]
+  (let [query (get-query component state-map)]
+    (if (fdn/possibly-stale? state-map prior-props)
+      (fdn/traced-db->tree state-map ident query)
+      prior-props)))
+
+(defn component-exists?
```

Comment:
Not an ideal name, but I'm not sure what to call it. There's already a bit of naming screw-up around the word "component" peppered throughout the codebase, so I'd prefer not to make that worse :D   I'll give it a bit of thought. Suggestions?

Reply:
I agree. Some ideas from the top of the head:

- loaded?
- loaded-state?
- has-some-state?
- some-state?
- state-exists?
- occupied?
- ident-occupied?
- in-use?

`has-active-state?` is already good and descriptive though.

I now wonder whether this is the right function, or if it is even necessary.

Is the `map?` check necessary? Or would it be sufficient to simply use `(seq (get-in state-map ident))`?

Reply:
Yeah, has-active-state? is fine by me.  And yes, should probably check it is a non-empty map so things don't get confused. That's the only way it will actually work.

Reply:
Very well, I have changed the name to `has-active-state?`